bagit.go

```diff
@@ -58,39 +59,56 @@ func NewBagIt() (*BagIt, error) {
 	return b, nil
 }
 
-func (b *BagIt) Validate(path string) error {
+// create a Python intepreter running the bagit-python wrapper.
+func (b *BagIt) create() (_ *runnerInstance, err error) {
+	i := &runnerInstance{}
+
 	cmd, err := b.ep.PythonCmd(filepath.Join(b.runner.GetExtractedPath(), "main.py"))
 	if err != nil {
-		return fmt.Errorf("create command: %v", err)
+		return nil, fmt.Errorf("create command: %v", err)
 	}
-	cmd.Stderr = os.Stderr
+	i.cmd = cmd
 
-	stdout, err := cmd.StdoutPipe()
+	stdin, err := cmd.StdinPipe()
 	if err != nil {
-		return fmt.Errorf("create stdout pipe: %v", err)
+		return nil, fmt.Errorf("create stdin pipe: %v", err)
 	}
-	defer stdout.Close()
+	i.stdin = stdin
 
-	stdin, err := cmd.StdinPipe()
+	stdout, err := cmd.StdoutPipe()
 	if err != nil {
-		return fmt.Errorf("create stdin pipe: %v", err)
+		return nil, fmt.Errorf("create stdout pipe: %v", err)
 	}
-	defer stdin.Close()
+	i.stdout = stdout
 
 	err = cmd.Start()
```

Comment:
It took me a while to figure out how the command runner was working, but I think I get it now.  It looks to me like `cmd.Start()` runs `main.py` in an `exec.Cmd` session then listens for input `i.stdin`.  Is there an advantage to keeping the command running, rather than just calling `cmd.Run()` like the example at https://github.com/kluctl/go-embed-python?  

It doesn't look to me like the cmd session is ever closed to release the system resources.  Did you mean to add a call to `i.stop()` somewhere here?

Reply:
I think that this is constrained by the fact that the [runner](https://github.com/artefactual-labs/bagit-gython/blob/main/internal/runner/main.py) runs a while loop to receive commands like `validate` or `make`. Given this, `cmd.Start()` simplifies things easier for the Go library because it's async (non-blocing), allowing me to start the runner and then send my command(s).

In practice, the Go library currently launches a new runner each time. With some additional effort, we could make the cmd session persist beyond a single method call. This would be advantageous because Python apps have a slow startup time, which can be problematic when validating hundreds of bags.